daemon/api_blob.go

```diff
@@ -0,0 +1,67 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package daemon
+
+import (
+	"net/http"
+
+	"github.com/snapcore/snapd/overlord/auth"
+	"github.com/snapcore/snapd/overlord/snapstate"
+	"github.com/snapcore/snapd/overlord/state"
+	"github.com/snapcore/snapd/snap"
+)
+
+var snapBlobCmd = &Command{
+	Path:     "/v2/snaps/{name}/blob",
+	UserOK:   true,
```

Comment:
The reasoning is that a user can already access the data because it's mounted, so this isn't an info leak.
The 0600 is because there could be sensitive data there, in a snap that's disabled (think a vendor core device where a user needs to pay to enable a premium snap, for example).

Reply:
should we call it blob?   other options:
* file
* snapfile
* content

Reply:
Download help message says:
```
The download command downloads the given snap and its supporting assertions
```
Then install help message:
```
Use --name to set the instance name when installing from snap file.
```
Throughout the code there are some references to 'blob' throughout the code, but mostly in tests.

`content` sounds fine too, though I have a feeling it might be understood as a way to get uncompressed snap content.

Reply:
My preference ATM would be "file"   (we are under /snaps/ so "snapfile" seems a bit redundant)